core/audits/is-on-https.js

```diff
@@ -74,6 +74,7 @@ class HTTPS extends Audit {
     const devtoolsLogs = artifacts.devtoolsLogs[Audit.DEFAULT_PASS];
     const networkRecords = await NetworkRecords.request(devtoolsLogs, context);
     const insecureURLs = networkRecords
+      .filter(record => !record.redirectDestination)
```

Comment:
This should be limited to the initial document request chain and not allow any other resources that redirected.

Reply:
Also don't want to allow `https` -> `http` -> `https` redirects?

Reply:
> This should be limited to the initial document request chain and not allow any other resources that redirected.

sgtm

> Also don't want to allow https -> http -> https redirects?

The impl in https://github.com/GoogleChrome/lighthouse/pull/13548 doesn't seem to care about intermediate redirects. Is there a strong reason for this?

Reply:
> sgtm, but should we allow iframe documents as well or just the main document?

IMO the only leeway should be if you put in an http URL to test, so not including iframes, yeah. Like you can't control what URL a user starts with (just redirect them), but you can control the URLs of any iframes the page loads.

> > Also don't want to allow https -> http -> https redirects?
> 
> The impl in #13548 doesn't seem to care about intermediate redirects. Is there a strong reason for this?

Yeah, not included there, and I added the question mark because I'm not sure if it's worth failing on. I was just trying to think through possible failure modes.

Seems bad? Starting secure but letting an intermediate response do any number of things that could be intercepted/altered. OTOH is there any increased risk vs starting with an insecure request? OTOOH, kind of ridiculous if that's the redirect chain for your site?

Reply:
> Starting secure but letting an intermediate response do any number of things that could be intercepted/altered.

This has me leaning more towards https://github.com/GoogleChrome/lighthouse/pull/13548 and closing this PR haha. This sounds like a real security problem that we would want to flag in any situation. It's just a *more severe* problem if the site didn't at least redirect to https.

Reply:
Haha, well, up to you :)

Couldn't resist checking, and there are about 100k/0.4% of sites in a recent HTTP Archive crawl that did this (went at least https->http->https, ending on an https).

Interestingly ~99% of them end up on the same origin they started on by the end of things. A lot of them are of the form
- `https://example.com`
- `http://example.com/en`
- `https://example.com/en`

so it does seem like a real error and something they can/should fix